src/parser/jobs/ast/Lightspeed.js

```diff
@@ -0,0 +1,177 @@
+import {Trans, i18nMark} from '@lingui/react'
+import React, {Fragment} from 'react'
+import {Accordion} from 'semantic-ui-react'
+
+import {ActionLink} from 'components/ui/DbLink'
+import Rotation from 'components/ui/Rotation'
+import ACTIONS, {getAction} from 'data/ACTIONS'
+import STATUSES from 'data/STATUSES'
+import Module from 'parser/core/Module'
+// import {Suggestion, TieredSuggestion, SEVERITY} from 'parser/core/modules/Suggestions'
+
+import DISPLAY_ORDER from './DISPLAY_ORDER'
+
+const LIGHTSPEED_CAST_TIME_MOD = -2.5
+// const LIGHTSPEED_LENGTH = 10000
+
+export default class LIGHTSPEED extends Module {
+	static handle = 'lightspeed'
+	static i18n_id = i18nMark('ast.lightspeed.title')
+	static dependencies = [
+		'castTime',
+		// 'suggestions',
+	]
+	static title = 'Lightspeed'
+	static displayOrder = DISPLAY_ORDER.LIGHTSPEED
+
+	_active = false
+	_lightspeed = {}
+	_history = []
+
+	_ctIndex = null
+
+	_missedGcds = 0
+
+	constructor(...args) {
+		super(...args)
+
+		this.addHook('cast', {by: 'player'}, this._onCast)
+
+		const lsBuffFilter = {
+			by: 'player',
+			abilityId: STATUSES.LIGHTSPEED.id,
+		}
+		this.addHook('applybuff', lsBuffFilter, this._onApplyLightspeed)
+		this.addHook('refreshbuff', lsBuffFilter, this._onRefreshLightspeed)
+		this.addHook('removebuff', lsBuffFilter, this._onRemoveLightspeed)
+
+		this.addHook('complete', this._onComplete)
+	}
+
+	_onCast(event) {
```

Comment:
I recognise that this is precisely what `smn/DWT` does, but I'm beginning to question starting the buff at the cast. Crap like swiftcast taking too long to apply and the next gcd still being a hardcast are an unfortunately common occurrence.

Not blocking, but worth considering.

Reply:
Maybe start from the timestamp of the buff? I could even get a suggestion out to tell them not to mash too much if we see  `cast` `hardcaststart` `applybuff` in the order. But I have PTSD from logs out of order in `ArcanaTracking.js` so i didn't want to put too much thought into it.